gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/ReferralRequestMapperTest.java

```diff
@@ -442,6 +460,83 @@ public void When_MapToReferralRequest_With_UnexpectedPriorityCode_Expect_Priorit
             .hasSize(1);
     }
 
+    @Test
+    void When_MapToReferralRequest_With_NopatConfidentialityCode_Expect_MetaFromConfidentialityServiceWithSecurity() {
+        final Meta metaWithSecurity = MetaFactory.getMetaFor(META_WITH_SECURITY, META_PROFILE);
+        final RCMRMT030101UKEhrExtract ehrExtract =
+            unmarshallEhrExtractElement("full_valid_data_example_with_nopat_confidentiality_code.xml");
+        final RCMRMT030101UKEhrComposition ehrComposition = GET_EHR_COMPOSITION.apply(ehrExtract);
+
+        when(confidentialityService.createMetaAndAddSecurityIfConfidentialityCodesPresent(
+            eq(META_PROFILE),
+            confidentialityCodeCaptor.capture(),
+            confidentialityCodeCaptor.capture()
+        )).thenReturn(metaWithSecurity);
+
+        final ReferralRequest result = mapReferralRequest(ehrExtract, ehrComposition,
+            composition -> composition.getComponent().get(0).getRequestStatement());
+
+        final CV confidentialityCode = confidentialityCodeCaptor
+            .getAllValues()
+            .get(0) // First value
+            .orElseThrow();
+
+        assertAll(
+            () -> assertThat(result.getMeta()).usingRecursiveComparison().isEqualTo(metaWithSecurity),
+            () -> assertThat(confidentialityCode.getCode()).isEqualTo(NOPAT),
+            () -> assertThat(confidentialityCodeCaptor.getAllValues().get(1)).isNotPresent()
+        );
+    }
+
+    @Test
+    void When_MapToReferralRequest_With_NopatConfidentialityCodeWithinEhrComposition_Expect_MetaFromConfidentialityServiceWithSecurity() {
+        final Meta metaWithSecurity = MetaFactory.getMetaFor(META_WITH_SECURITY, META_PROFILE);
+        final RCMRMT030101UKEhrExtract ehrExtract =
+            unmarshallEhrExtractElement("full_valid_data_example_with_nopat_confidentiality_code_in_ehr_composition.xml");
+        final RCMRMT030101UKEhrComposition ehrComposition = GET_EHR_COMPOSITION.apply(ehrExtract);
+
+        when(confidentialityService.createMetaAndAddSecurityIfConfidentialityCodesPresent(
+            eq(META_PROFILE),
+            confidentialityCodeCaptor.capture(),
+            confidentialityCodeCaptor.capture()
+        )).thenReturn(metaWithSecurity);
+
+        final ReferralRequest result = mapReferralRequest(ehrExtract, ehrComposition,
+            composition -> composition.getComponent().get(0).getRequestStatement());
+
+        final CV confidentialityCode = confidentialityCodeCaptor
+            .getAllValues()
+            .get(1) // Second value (i.e. ehrComposition.getConfidentialityCode())
+            .orElseThrow();
+
+        assertAll(
+            () -> assertThat(result.getMeta()).usingRecursiveComparison().isEqualTo(metaWithSecurity),
+            () -> assertThat(confidentialityCode.getCode()).isEqualTo(NOPAT),
+            () -> assertThat(confidentialityCodeCaptor.getAllValues().get(0)).isNotPresent()
+        );
+    }
```

Comment:
For these tests, could we use a parameterized test where the argument being passed to the test is the XML file name?